Sources/FoundationExtensions/LossyCollections.swift

```diff
@@ -0,0 +1,221 @@
+/// but this wrapper can be composed with `@DefaultDecodable.EmptyArray`
+/// to make it produce an empty array in case of any other type error.
+/// ```
+/// struct Data: Decodable {
+///      @DefaultDecodable.EmptyArray @LossyArray var list: [Int]
+/// }
+/// ```
+@propertyWrapper
+struct LossyArray<Value> {
+
+    var wrappedValue: [Value]
+
+    init(wrappedValue: [Value]) { self.wrappedValue = wrappedValue }
+
+}
+
+extension LossyArray: Decodable where Value: Decodable {
+
+    private struct AnyDecodableValue: Decodable {}
+
+    init(from decoder: Decoder) throws {
+        var container = try decoder.unkeyedContainer()
+
+        var elements: [Value] = []
+        while !container.isAtEnd {
+            do {
+                elements.append(try container.decode(Value.self))
+            } catch {
+                do {
+                    // Attempt to decode anything to skip this element
+                    _ = try container.decode(AnyDecodableValue.self)
+                } catch {
+                    ErrorUtils.logDecodingError(error)
+                }
+            }
+        }
+
+        self.wrappedValue = elements
+    }
+
+}
+
+extension LossyArray: Encodable where Value: Encodable {
+
+    func encode(to encoder: Encoder) throws {
+        try wrappedValue.encode(to: encoder)
+    }
+
+}
+
+extension LossyArray: Equatable where Value: Equatable { }
+extension LossyArray: Hashable where Value: Hashable { }
+
+extension LossyArray: ExpressibleByArrayLiteral {
+
+    init(arrayLiteral elements: Value...) {
+        self.wrappedValue = elements
+    }
+
+}
+
+// MARK: - LossyDictionary
+
+/// A property wrapper that allows decoding `Dictionary`s and ignore elements that fail to decode.
+/// For example, this will ignore any elements that aren't numbers, instead of failing to decode altogether.
+/// ```
+/// struct Data: Decodable {
+///     @LossyDictionary var map: [String: Int]
+/// }
+/// ```
+/// This does require that the value is a `Dictionary`,
+/// but this wrapper can be composed with `@DefaultDecodable.EmptyDictionary`
+/// to make it produce an empty dictionary in case of any other type error.
+/// ```
+/// struct Data: Decodable {
+///      @DefaultDecodable.EmptyDictionary @LossyDictionary var map: [String: Int]
+/// }
+/// ```
+/// - Note: if the values of the dictionary are an array, consider using `LossyArrayDictionary`.
+@propertyWrapper
+struct LossyDictionary<Value: Decodable> {
+
+    var wrappedValue: [String: Value]
+
+    init(wrappedValue: [String: Value]) {
+        self.wrappedValue = wrappedValue
+    }
+
+}
+
+/// A property wrapper that allows decoding `Dictionary`s of `Array`s and ignore elements that fail to decode.
+/// This is a composition of `LossyArray` and `LossyDictionary`, that due to limitations of property wrappers,
+/// it necessitates its own type.
+@propertyWrapper
+struct LossyArrayDictionary<Value: Decodable> {
+
+    var wrappedValue: [String: [Value]]
```

Comment:
Is it safe to assume the key will always be a `String`? I mean, I can't see why it wouldn't be in any of our cases but not sure if that something we want to open up 🤷‍♂️

Reply:
Good question. Swift's [`CodingKey`](https://developer.apple.com/documentation/swift/codingkey) allows keys to be `String` or `Int`, in a way that Integer keys can be implicitly converted to `String` when decoding JSON dictionaries. We don't use that, so for simplicity these property wrappers assume just `String`, but it would be _trivial_ to change these by adding a `CodingKey` `associatedtype` so that integer keys are supported as well.